src/java.se/share/data/jdwp/jdwp.spec

```diff
@@ -1168,6 +1168,15 @@ JDWP "Java(tm) Debug Wire Protocol"
         "the single threaded invocation will not be suspended when the "
         "invocation completes. "
         "<p>"
+        "<a href=../../api/java.base/java/lang/Thread.html#virtual-threads>Virtual threads</a> "
+        "are typically user-mode threads scheduled by the Java runtime rather than "
+        "the operating system. As such, there may be more cases with virtual threads "
+        "where the INVOKE_SINGLE_THREADED option may cause a deadlock. "
+        "For example, suspending all threads in the target VM may suspend threads that "
+        "support the timer mechanism for virtual threads, and thus methods such as "
+        "<a href=../../api/java.base/java/lang/Thread.html#sleep(long)>Thread.sleep(long)</a> "
+        "may deadlock."
```

Comment:
I think this is an implementation detail that you don't really want to document in these unrelated specifications. I'm also unclear why virtual threads are more of a problem here than regular threads, if you have suspended them all - a regular thread won't return from sleep until it is unsuspended. In fact the more I think about this, the less I see why virtual threads need special mention here at all.

Reply:
Of course as soon as I posted that I now realize the issue. There are more things that could "deadlock" virtual threads when all other threads are suspended - such as calling sleep, because it requires a "system" thread to be active. This suggest to me that we need modifications to JDWP/JDI to support effective debugging of virtual threads, by allowing for the notion of "system threads" that by default are not suspended (or else always resumed for an invoke).

Reply:
A general point here is that there many cases that are unrelated to virtual threads where suspending all, except one, thread will deadlock/hang if other threads can't make progress. This includes several standard (java.*) APIs.

It is true that there are a couple of additional blocking operations with virtual threads that require I/O or timer threads to make progress in order to avoid a virtual thread being parked or blocked indefinitely. It feels more like something for  "@apiNote" for the invokeMethod/etc. javadoc, not clear to me that it's worth adding as a note in the description of the commands in the JDWP spec.

Reply:
I can change it to `@apiNote`. Since it is not really part of the spec, I can see leaving it out of the JDWP spec. It's mostly important to get this info to JDI users.

Reply:
> I can change it to `@apiNote`. Since it is not really part of the spec, I can see leaving it out of the JDWP spec. It's mostly important to get this info to JDI users.

Okay, in which case a CSR is not strictly needed here (I added the "csr" label because of the spec update).